[2025-06-13T19:09:04.860+0200] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-13T17:08:58.746389+00:00 [queued]>
[2025-06-13T19:09:04.881+0200] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-13T17:08:58.746389+00:00 [queued]>
[2025-06-13T19:09:04.881+0200] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2025-06-13T19:09:04.881+0200] {taskinstance.py:1280} INFO - Starting attempt 1 of 4
[2025-06-13T19:09:04.882+0200] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2025-06-13T19:09:04.905+0200] {taskinstance.py:1300} INFO - Executing <Task(BashOperator): pyspark_train_classifier_model> on 2025-06-13 17:08:58.746389+00:00
[2025-06-13T19:09:04.908+0200] {standard_task_runner.py:55} INFO - Started process 104573 to run task
[2025-06-13T19:09:04.914+0200] {standard_task_runner.py:82} INFO - Running: ['airflow', 'tasks', 'run', 'agile_data_science_batch_prediction_model_training', 'pyspark_train_classifier_model', 'manual__2025-06-13T17:08:58.746389+00:00', '--job-id', '63', '--raw', '--subdir', 'DAGS_FOLDER/setup.py', '--cfg-path', '/tmp/tmpci0l_xti']
[2025-06-13T19:09:04.916+0200] {standard_task_runner.py:83} INFO - Job 63: Subtask pyspark_train_classifier_model
[2025-06-13T19:09:05.023+0200] {task_command.py:388} INFO - Running <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-13T17:08:58.746389+00:00 [running]> on host l089.lab.dit.upm.es
[2025-06-13T19:09:05.138+0200] {taskinstance.py:1507} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=agile_data_science_batch_prediction_model_training
AIRFLOW_CTX_TASK_ID=pyspark_train_classifier_model
AIRFLOW_CTX_EXECUTION_DATE=2025-06-13T17:08:58.746389+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2025-06-13T17:08:58.746389+00:00
[2025-06-13T19:09:05.140+0200] {subprocess.py:63} INFO - Tmp dir root location: 
 /tmp
[2025-06-13T19:09:05.141+0200] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\nspark-submit --master local[4]   /home/monica.fernandez/practica_creativa//resources/train_spark_mllib_model.py   /home/monica.fernandez/practica_creativa/']
[2025-06-13T19:09:05.150+0200] {subprocess.py:86} INFO - Output:
[2025-06-13T19:09:09.036+0200] {subprocess.py:93} INFO - 25/06/13 19:09:09 WARN Utils: Your hostname, l089 resolves to a loopback address: 127.0.1.1; using 138.4.31.89 instead (on interface enp3s0)
[2025-06-13T19:09:09.040+0200] {subprocess.py:93} INFO - 25/06/13 19:09:09 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2025-06-13T19:09:15.814+0200] {subprocess.py:93} INFO - 25/06/13 19:09:15 INFO SparkContext: Running Spark version 3.5.3
[2025-06-13T19:09:15.818+0200] {subprocess.py:93} INFO - 25/06/13 19:09:15 INFO SparkContext: OS info Linux, 6.1.0-32-amd64, amd64
[2025-06-13T19:09:15.823+0200] {subprocess.py:93} INFO - 25/06/13 19:09:15 INFO SparkContext: Java version 17.0.14
[2025-06-13T19:09:16.009+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-06-13T19:09:16.466+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO ResourceUtils: ==============================================================
[2025-06-13T19:09:16.466+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-06-13T19:09:16.467+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO ResourceUtils: ==============================================================
[2025-06-13T19:09:16.482+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO SparkContext: Submitted application: train_spark_mllib_model.py
[2025-06-13T19:09:16.512+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-06-13T19:09:16.527+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO ResourceProfile: Limiting resource is cpu
[2025-06-13T19:09:16.539+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-06-13T19:09:16.784+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO SecurityManager: Changing view acls to: monica.fernandez
[2025-06-13T19:09:16.788+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO SecurityManager: Changing modify acls to: monica.fernandez
[2025-06-13T19:09:16.798+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO SecurityManager: Changing view acls groups to:
[2025-06-13T19:09:16.809+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO SecurityManager: Changing modify acls groups to:
[2025-06-13T19:09:16.822+0200] {subprocess.py:93} INFO - 25/06/13 19:09:16 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: monica.fernandez; groups with view permissions: EMPTY; users with modify permissions: monica.fernandez; groups with modify permissions: EMPTY
[2025-06-13T19:09:17.574+0200] {subprocess.py:93} INFO - 25/06/13 19:09:17 INFO Utils: Successfully started service 'sparkDriver' on port 38421.
[2025-06-13T19:09:17.687+0200] {subprocess.py:93} INFO - 25/06/13 19:09:17 INFO SparkEnv: Registering MapOutputTracker
[2025-06-13T19:09:17.761+0200] {subprocess.py:93} INFO - 25/06/13 19:09:17 INFO SparkEnv: Registering BlockManagerMaster
[2025-06-13T19:09:17.793+0200] {subprocess.py:93} INFO - 25/06/13 19:09:17 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-06-13T19:09:17.794+0200] {subprocess.py:93} INFO - 25/06/13 19:09:17 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-06-13T19:09:17.815+0200] {subprocess.py:93} INFO - 25/06/13 19:09:17 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-06-13T19:09:17.867+0200] {subprocess.py:93} INFO - 25/06/13 19:09:17 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-f0eb165d-fdd5-4364-8eef-3575ab660731
[2025-06-13T19:09:17.913+0200] {subprocess.py:93} INFO - 25/06/13 19:09:17 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-06-13T19:09:17.948+0200] {subprocess.py:93} INFO - 25/06/13 19:09:17 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-06-13T19:09:18.128+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-06-13T19:09:18.258+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2025-06-13T19:09:18.268+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO Utils: Successfully started service 'SparkUI' on port 4041.
[2025-06-13T19:09:18.512+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO Executor: Starting executor ID driver on host 138.4.31.89
[2025-06-13T19:09:18.512+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO Executor: OS info Linux, 6.1.0-32-amd64, amd64
[2025-06-13T19:09:18.512+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO Executor: Java version 17.0.14
[2025-06-13T19:09:18.535+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO Executor: Starting executor with user classpath (userClassPathFirst = false): ''
[2025-06-13T19:09:18.540+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO Executor: Created or updated repl class loader org.apache.spark.util.MutableURLClassLoader@5de43805 for default.
[2025-06-13T19:09:18.607+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 38773.
[2025-06-13T19:09:18.607+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO NettyBlockTransferService: Server created on 138.4.31.89:38773
[2025-06-13T19:09:18.610+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-06-13T19:09:18.617+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 138.4.31.89, 38773, None)
[2025-06-13T19:09:18.624+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO BlockManagerMasterEndpoint: Registering block manager 138.4.31.89:38773 with 434.4 MiB RAM, BlockManagerId(driver, 138.4.31.89, 38773, None)
[2025-06-13T19:09:18.627+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 138.4.31.89, 38773, None)
[2025-06-13T19:09:18.631+0200] {subprocess.py:93} INFO - 25/06/13 19:09:18 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 138.4.31.89, 38773, None)
[2025-06-13T19:09:19.742+0200] {subprocess.py:93} INFO - 25/06/13 19:09:19 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-06-13T19:09:19.760+0200] {subprocess.py:93} INFO - 25/06/13 19:09:19 INFO SharedState: Warehouse path is 'file:/tmp/airflowtmpqdlpov9f/spark-warehouse'.
[2025-06-13T19:09:23.205+0200] {subprocess.py:93} INFO - MLflow Run ID: 9b34211a42d8473ca2a6d392fa9c2956, Experiment ID: 0
[2025-06-13T19:09:23.205+0200] {subprocess.py:93} INFO - MLflow Tracking URI: file:///tmp/airflowtmpqdlpov9f/mlruns
[2025-06-13T19:09:24.174+0200] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-06-13T19:09:24.174+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/resources/train_spark_mllib_model.py", line 411, in <module>
[2025-06-13T19:09:24.174+0200] {subprocess.py:93} INFO -     main(sys.argv[1])
[2025-06-13T19:09:24.174+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/resources/train_spark_mllib_model.py", line 111, in main
[2025-06-13T19:09:24.175+0200] {subprocess.py:93} INFO -     features = spark.read.json(input_path, schema=schema)
[2025-06-13T19:09:24.175+0200] {subprocess.py:93} INFO -                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-06-13T19:09:24.175+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/.sdkman/candidates/spark/current/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 425, in json
[2025-06-13T19:09:24.175+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/.sdkman/candidates/spark/current/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-06-13T19:09:24.175+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/.sdkman/candidates/spark/current/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2025-06-13T19:09:24.203+0200] {subprocess.py:93} INFO - pyspark.errors.exceptions.captured.AnalysisException: [PATH_NOT_FOUND] Path does not exist: file:/home/monica.fernandez/data/simple_flight_delay_features.jsonl.bz2.
[2025-06-13T19:09:24.574+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO SparkContext: Invoking stop() from shutdown hook
[2025-06-13T19:09:24.575+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-06-13T19:09:24.597+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO SparkUI: Stopped Spark web UI at http://138.4.31.89:4041
[2025-06-13T19:09:24.642+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-06-13T19:09:24.682+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO MemoryStore: MemoryStore cleared
[2025-06-13T19:09:24.684+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO BlockManager: BlockManager stopped
[2025-06-13T19:09:24.697+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-06-13T19:09:24.705+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-06-13T19:09:24.717+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO SparkContext: Successfully stopped SparkContext
[2025-06-13T19:09:24.718+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO ShutdownHookManager: Shutdown hook called
[2025-06-13T19:09:24.719+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO ShutdownHookManager: Deleting directory /tmp/spark-2cb9d684-ae5c-4c51-88c9-6ecb9af8917a
[2025-06-13T19:09:24.725+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO ShutdownHookManager: Deleting directory /tmp/spark-56d42dad-f110-4a67-b869-ab1b6e4b001f/pyspark-64eb851c-edc3-4fbc-9c27-1e8b1719623e
[2025-06-13T19:09:24.733+0200] {subprocess.py:93} INFO - 25/06/13 19:09:24 INFO ShutdownHookManager: Deleting directory /tmp/spark-56d42dad-f110-4a67-b869-ab1b6e4b001f
[2025-06-13T19:09:24.844+0200] {subprocess.py:97} INFO - Command exited with return code 1
[2025-06-13T19:09:24.872+0200] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/airflow/operators/bash.py", line 196, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-06-13T19:09:24.879+0200] {taskinstance.py:1318} INFO - Marking task as UP_FOR_RETRY. dag_id=agile_data_science_batch_prediction_model_training, task_id=pyspark_train_classifier_model, execution_date=20250613T170858, start_date=20250613T170904, end_date=20250613T170924
[2025-06-13T19:09:24.917+0200] {standard_task_runner.py:100} ERROR - Failed to execute job 63 for task pyspark_train_classifier_model (Bash command failed. The command returned a non-zero exit code 1.; 104573)
[2025-06-13T19:09:24.955+0200] {local_task_job.py:208} INFO - Task exited with return code 1
[2025-06-13T19:09:24.990+0200] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
